UITests/FullScreenViewSpec.swift

```diff
@@ -0,0 +1,211 @@
+import XCTest
+import Quick
+import Nimble
+
+class FullScreenViewSpec: QuickSpec {
+
+    override func spec() {
+
+        var app: XCUIApplication!
+        var iamView: XCUIElement {
+            app.otherElements.element(matching: NSPredicate(format: "identifier BEGINSWITH[cd] 'IAMView'"))
+        }
+
+        func launchAppIfNecessary(args: String) {
+            guard app == nil || !app.launchArguments.joined(separator: " ").contains(args) else {
+                return
+            }
+            self.continueAfterFailure = false
+            app = XCUIApplication()
+            app.launchArguments.append("--uitesting")
+            app.launchArguments.append(args)
+            app.launch()
+        }
+
+        describe("Full-screen campaign view") {
+
+            context("when clicking X button") {
+
+                beforeEach {
+                    launchAppIfNecessary(args: "-campaignType full-text-only")
+                    if !iamView.exists {
+                        app.buttons["login_successful"].tap()
+                        expect(iamView.exists).toEventually(beTrue())
+                    }
+                }
+
+                it("should close the campaign") {
+                    iamView.buttons["exitButton"].tap()
+                    expect(iamView.exists).to(beFalse())
+                }
+
+                it("should have 44pt touch area") {
+                    let exitButtonCenter = iamView.buttons["exitButton"].coordinate(withNormalizedOffset: CGVector(dx: 0.5, dy: 0.5))
+                    let upperLeftCorner = exitButtonCenter.withOffset(CGVector(dx: -22, dy: -22))
+                    upperLeftCorner.tap()
+                    expect(iamView.exists).to(beFalse())
+
+                    app.buttons["login_successful"].tap() // show the message again
+                    expect(iamView.exists).toEventually(beTrue())
+                    let bottomRightCorner = exitButtonCenter.withOffset(CGVector(dx: 21, dy: 21)) // points on max edges are not counted
+                    bottomRightCorner.tap()
+                    expect(iamView.exists).to(beFalse())
+                }
+            }
+
+            context("without controls") {
+
+                beforeEach {
+                    launchAppIfNecessary(args: "-campaignType full-text-only")
+                    if !iamView.exists {
+                        app.buttons["login_successful"].tap()
+                        expect(iamView.exists).toEventually(beTrue())
+                    }
+                }
+
+                it("should not display any buttons") {
+                    expect(iamView.buttons["Button0"].exists).to(beFalse())
+                    expect(iamView.buttons["Button1"].exists).to(beFalse())
+                }
+
+                it("should not display opt-out checkbox") {
+                    expect(iamView.buttons["Do not show me this message again"].exists).to(beFalse())
```

Comment:
Will this work if the device is set to Japanese or are we forcing EN locale for our test env?

Reply:
Tests always run in english. But actually, I can add an identifier and use it instead